src/render/renderable_repo.cpp

```diff
@@ -20,7 +20,13 @@ bool RenderableRepo::AddSpritesFromOneTextureDesc( Json::Value& TexDesc, Element
     {
         return false;
     }
-    int32_t TexId = AutoId( boost::filesystem::path( PathStr ).generic_string() );
+    auto path = boost::filesystem::path( PathStr );
+    if (!path.has_parent_path())
+    {
+        path = parentPath;
```

Comment:
This way you can no longer specify textures at the root dir. I'd suggest using the "standard" absolute/relative paths specifications, if the path is absolute, use that, if path is relative, use the parentPath / path. ( yes, that requires changing all the existing path specs in the jsons,
`sed 's#\(\s*"texture_path"\s*:\s*"\)\([a-zA-Z0-9]\{1,\}/.\{1,\}"\)\(.*\)#\1/\2\3#g'` does that for you, seems okay for the jsons directly in the sprites/ folder )